src/pytrajplot/parse_data.py

```diff
@@ -491,6 +491,10 @@ def check_input_dir(input_dir, prefix_dict, separator):
         ):  # if filename ≡ plot_info filename
             plot_info_dict = PLOT_INFO(file=file_path).data
 
+    # check if plot_info has been found
+    if not "plot_info_dict" in dir():
+        raise RuntimeError("ERROR: plot_info file not found:", prefix_dict["plot_info"])
+
```

Comment:
```suggestion
            plot_info_dict = PLOT_INFO(file=file_path).data
            break
    else:
        raise RuntimeError("ERROR: plot_info file not found:", prefix_dict["plot_info"])

```
The first match should be sufficient and therefore the loop can be exited. Additionally this removes the need for `"plot_info_dict" in dir()`.

Reply:
I'm pretty sure it cannot be solved like this. Printing os.listdir(input_dir) provides an unordered list of filenames--> ['plot_info', 'startf_006-144F', 'tra_geom_006-144F']. So, the loop would be terminated too early in this case.
What about using the good ol' None type?

`
if plot_info_dict is None:
        raise RuntimeError("ERROR: plot_info file not found:", prefix_dict["plot_info"])
`

Reply:
Oh yes, I see. Some asserts would be skipped and this might be unexpected.
It's still a weird function that's named "check" but checks stuff and retrieves stuff. But that's not subject to this PR.
I like your idea with `None`.